File: plumbing_pub_sub/src/demo_sub_person.cpp
#define BOOST_BIND_GLOBAL_PLACEHOLDERS

#include "ros/ros.h"
#include "plumbing_pub_sub/Person.h"

void doPerson(const plumbing_pub_sub::Person::ConstPtr& person)
{
    ROS_INFO("Name %s, Age %d, Height %.2f", person->name.c_str(),person->age,person->height);
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "jiazhang");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("liaotian", 10, doPerson);

    ros::spin();
    return 0;
}

File: plumbing_pub_sub/scripts/demo_sub_person.py
#! /usr/bin/env python

import rospy
from plumbing_pub_sub.msg import Person

def doPerson(p):
    rospy.loginfo("Recevied: %s,%d,%.2f", p.name,p.age,p.height)

if __name__ == "__main__":
    rospy.init_node("daYe")
    sub = rospy.Subscriber("tan",Person,doPerson,queue_size=10)
    rospy.spin()
